server.js

```diff
@@ -65,7 +65,7 @@ app.prepare().then(() => {
         return app.render(req, res, `/newsletter/${staticSlug}`, req.query)
       })
     })
-  server.get('/newsletter/:slug', (req, res) => {
+  server.get('/newsletter/:year(\\d{4})/:month(\\d{2})/:day(\\d{2})/:slug', (req, res) => {
```

Comment:
Couldn't we do something like this and keep the old routes as is?
```suggestion
  server.get('/newsletter/:path*', (req, res) => {
```